manager/manager.go

```
@@ -0,0 +1,117 @@
+	queue      dispatch.Q
+}
+
+type Listener interface {
+	Update(items []argus.Item)
+}
+
+type EventSender interface {
+	Send(event *wrp.Message, deviceID string) //send event to all dispatchers in map
+}
+
+func (m *Manager) Update(items []argus.Item) {
+	recentMap := make(map[string]model.Norn)
+	newNorns := []nornDispatcher{}
+	oldNorns := []dispatch.D{}
+
+	for _, item := range items {
+		id := item.Identifier
+		norn, err := norn.ConvertItemToNorn(item)
+		if err != nil {
+			log.WithPrefix(m.logger, level.Key(), level.ErrorValue()).Log(logging.MessageKey(), "failed to convert Item to Norn", "item", item)
+		}
+		if _, ok := m.nornsDispatch[id]; ok {
+			recentMap[id] = norn
+		} else {
+			dispatcher, err := dispatch.NewDispatcher(m.dispatcherConfig)
+			if err == nil {
+				newNorns = append(newNorns, nornDispatcher{norn: norn, dispatcher: dispatcher})
+				recentMap[id] = norn
+			}
+		}
+	}
+
+	for i, norndis := range m.nornsDispatch {
+		if val, ok := recentMap[i]; !ok {
+			oldNorns = append(oldNorns, norndis.dispatcher)
+			delete(m.nornsDispatch, i)
+		} else {
+			// update TTL for norn
+			norndis.dispatcher.Update(val)
+		}
+	}
+
+	for _, norndis := range newNorns {
+		m.nornsDispatch[norndis.norn.DeviceID] = norndis
+	}
+
+	for _, dispatcher := range oldNorns {
+		dispatcher.Stop()
+	}
+
+}
+
+func (m *Manager) Send(event *wrp.Message, deviceID string) {
+	for _, nd := range m.nornsDispatch {
+		// call Queue()
+		en := dispatch.NewEventNorn(event, nd.norn, deviceID)
+		nd.queue.Queue(en)
+		// nd.dispatcher.Queue(en)
+		// nd.dispatcher.Dispatch(event, nd.norn, deviceID)
+	}
+}
+
+// GET '/norns/id'
+func (m *Manager) GetNorn(rw http.ResponseWriter, req *http.Request) (norn model.Norn, err error) {
+	nornID := mux.Vars(req)
+	id := nornID["id"]
+
+	if norndis, ok := m.nornsDispatch[id]; ok {
+		return norndis.norn, nil
+	} else {
+		logging.Info(m.logger).Log(logging.MessageKey(), "Could not get norn", logging.ErrorKey(), err)
+	}
```

We need to add a response code here - we shouldn't return 200 when the norn doesn't exist.  I believe we want to return 404 Not Found.  That is, unless some other handler will return the 404 - then we should at least return an error.